db/wal_edit.h

```diff
@@ -30,6 +32,12 @@ class WalMetadata {
 
   explicit WalMetadata(uint64_t size_bytes) : size_bytes_(size_bytes) {}
 
+  bool IsClosed() const { return HasSize(); }
```

Comment:
Can you remind me why `IsClosed()` is equivalent to `HasSize()`?

Reply:
left inline comments

Reply:
If application calls `SyncWAL()` or writes with `WriteOptions.sync=true`, then the WAL will have part of it synced. Recovery should check the actual WAL file size against this.
Things can be more complex if we allow log file number recycling.

Reply:
We will only write size of the WAL after closing it.